sql/core/src/main/scala/org/apache/spark/sql/execution/local/ConvertToSafeNode.scala

```diff
@@ -27,6 +27,10 @@ case class ConvertToSafeNode(conf: SQLConf, child: LocalNode) extends UnaryLocal
 
   private[this] var convertToSafe: Projection = _
 
+  override def prepare(): Unit = {
+    child.prepare()
```

Comment:
Why not implement this one in `UnaryLocalNode`? Then we don't need to duplicate them in all subclasses of ``UnaryLocalNode`.